### DISCUSSION TITLE
thermal model

### DISCUSSION THREAD

**Discussion by anirbanroy16** [pybamm-team/PyBaMM#3448] [Q&A] [answered]
I notice the lumped model is created out of "coarsening" a DFN model. Is there any way we can obtain values for voltage gradient and overpotential without solving the DFN model first?

https://colab.research.google.com/github/pybamm-team/PyBaMM/blob/main/examples/notebooks/models/thermal-models.ipynb

**Comment by rtimms** [comment]
The lumped thermal model can be used with any electrochemical model and just gets the heat source terms through volume-averaging. What do you mean by "voltage gradient"? Some reduced-order models still give you distributed quantities at the electrode level (e.g. SPMe gives you gradients in electrolyte potential and concentration) - is this what you mean?

**Reply by anirbanroy16** [reply]
Thanks. My question was specifically - how do we get these terms in the screenshot without running a DFN model a-priori?

![image](https://github.com/pybamm-team/PyBaMM/assets/25733848/6f14e16e-0e02-424c-b851-9b418447a567)

**Reply by rtimms** [reply, marked as answer]
You need to solve an electrochemical model together with the thermal model to calculate the heat source terms.

**Reply by anirbanroy16** [reply]
Got it, thank you. How do we estimate the voltage gradient around each electrode using SPM/SPMe? any references/pointers are much appreciated.

**Reply by anirbanroy16** [reply]
My basic question is - if we have just IVT data and are running a SPM/SPMe, what is the best way to estimate the source terms?

**Reply by rtimms** [reply]
You calculate the source terms from the model using the expressions in your screenshot. I'm not sure I follow your question about estimating the source terms from data. Do you mean you want to fit parameters for SPM/SPMe/DFN, which will then allow you to compute the source terms?
